dotCMS/src/main/java/com/dotmarketing/portlets/workflows/business/WorkflowAPIImpl.java

```diff
@@ -204,6 +204,14 @@ public void saveSchemesForStruct(final Structure struc, final List<WorkflowSchem
 		}
 	}
 
+	public void saveSchemeIdsForStruct(final ContentType contentType, List<String> schemesIds) throws DotDataException {
+		try {
+			workFlowFactory.saveSchemeIdsForStruct(contentType.inode(), schemesIds);

```

Comment:
I am not pretty sure but I think you may want to find each scheme object in order to see if all schemes to associated to the contentType exists.
In addition you should not use the word "struct" if it is not a legacy method, it must be content type, structure is a legacy and non-longer-used word.

Reply:
You should add log here

Reply:
One more thing: this method should be annotated by wrapInTransaction

Reply:
Upd: checking the table def:
```
create table workflow_scheme_x_structure(
	id varchar(36) primary key,
	scheme_id varchar(36) references workflow_scheme(id),
	structure_id varchar(36) references structure(inode)
);
```
The reference is there, so may be not need to see if it is exists, however you should try to add a non-existing schemeId and see which kind of exception is returned and make the right Response/Status based on it.